patches/server/0973-Fix-NPE-on-Boat-getStatus.patch

```diff
@@ -0,0 +1,32 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: LemonCaramel <admin@caramel.moe>
+Date: Mon, 10 Apr 2023 20:48:26 +0900
+Subject: [PATCH] Fix NPE on Boat getStatus
+
+Boat status is null until the entity is added to the world and the tick() method is called.
+
+== AT ==
+public net.minecraft.world.entity.vehicle.Boat getStatus()Lnet/minecraft/world/entity/vehicle/Boat$Status;
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftBoat.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftBoat.java
+index dd7076938b04d4b36e3360a883bae81ced455fda..8d1fab1045e89dfa2f42714847288fe863b9530d 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftBoat.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftBoat.java
+@@ -89,6 +89,17 @@ public class CraftBoat extends CraftVehicle implements Boat {
+ 
+     @Override
+     public Status getStatus() {
++        // Paper start
++        final var handle = this.getHandle();
++        if (handle.status == null) {
++            if (handle.valid && !handle.updatingSectionStatus) {
++                handle.status = handle.getStatus();
++            } else {
++                return Status.NOT_IN_WORLD;
++            }
++        }
++        if (true) return CraftBoat.boatStatusFromNms(handle.status);
```

Comment:
Is this if (True) needed here? As isn't this just doing the same thing?

Reply:
I just didn't want to call getHandle twice.

Reply:
getHandle is just a cast so avoiding calling it twice isn't too important